commit 2465c30c3b11ee9436dc753642cb53bf4187ad43
Author: Tim Foster <tim.foster@joyent.com>
Date:   2019-03-11T17:30:01+00:00 (7 months ago)
    
    MANTA-4076 convert manta-garbage-collector to engbld framework
    Reviewed by: Richard Bradley <richard.bradley@joyent.com>
    Approved by: Kelly McLaughlin <kelly.mclaughlin@joyent.com>

diff --git a/.gitignore b/.gitignore
index e050ca7..a3ca455 100644
--- a/.gitignore
+++ b/.gitignore
@@ -1,7 +1,8 @@
 /node_modules
+/bits
 /tmp
 /cache/
-build
+/build
 /make_stamps
 docs/*.json
 docs/*.html
@@ -12,5 +13,4 @@ core*
 proto/
 man
 node
-*-pkg-*.tar.bz2
-smf/manifests/bapi.xml
+*-pkg-*.tar.gz
diff --git a/.gitmodules b/.gitmodules
index 4ccb4e9..db80d77 100644
--- a/.gitmodules
+++ b/.gitmodules
@@ -10,3 +10,6 @@
 [submodule "deps/jsstyle"]
 	path = deps/jsstyle
 	url = https://github.com/davepacheco/jsstyle.git
+[submodule "deps/eng"]
+	path = deps/eng
+	url = https://github.com/joyent/eng.git
diff --git a/Makefile b/Makefile
index f192017..da96a1f 100644
--- a/Makefile
+++ b/Makefile
@@ -26,17 +26,15 @@ PREFIX =			/opt/smartdc/$(NAME)
 
 CLEAN_FILES +=			$(PROTO)
 
-RELEASE_TARBALL =		$(NAME)-pkg-$(STAMP).tar.bz2
-
-
-include ./tools/mk/Makefile.defs
-include ./tools/mk/Makefile.node_prebuilt.defs
-include ./tools/mk/Makefile.node_modules.defs
-
-.PHONY: all
-all: $(STAMP_NODE_PREBUILT) $(STAMP_NODE_MODULES)
-	$(NODE) --version
+RELEASE_TARBALL =		$(NAME)-pkg-$(STAMP).tar.gz
 
+ENGBLD_USE_BUILDIMAGE =		true
+ENGBLD_REQUIRE :=		$(shell git submodule update --init deps/eng)
+include ./deps/eng/tools/mk/Makefile.defs
+TOP ?= $(error Unable to access eng.git submodule Makefiles.)
+include ./deps/eng/tools/mk/Makefile.node_prebuilt.defs
+include ./deps/eng/tools/mk/Makefile.agent_prebuilt.defs
+include ./deps/eng/tools/mk/Makefile.node_modules.defs
 
 #
 # Install macros and targets:
@@ -115,22 +113,38 @@ JSL_FILES_NODE = 		$(JS_FILES)
 
 JSSTYLE_FILES = 		$(JS_FILES)
 
-JSL_CONF_NODE = 		tools/jsl.node.conf
+JSL_CONF_NODE = 		deps/eng/tools/jsl.node.conf
 
 JSON_FILES = 			package.json
 
+#
+# Stuff used for buildimage
+#
+BASE_IMAGE_UUID		= 04a48d7d-6bb5-4e83-8c3b-e60a99e0f48f
+BUILDIMAGE_NAME		= $(NAME)
+BUILDIMAGE_DESC		= Manta Garbage Collector
+AGENTS = amon config registrar
+
+.PHONY: all
+all: $(STAMP_NODE_PREBUILT) $(STAMP_NODE_MODULES) install
+	$(NODE) --version
+
 .PHONY: test
 test: | $(CATEST)
 	$(CATEST) $(CATEST_FILES)
 
 $(CATEST): deps/catest/.git
 
+$(INSTALL_FILES): manta-scripts
+
 .PHONY: install
-install: $(INSTALL_FILES)
+install: $(NODE_EXEC) $(INSTALL_FILES)
 
 $(INSTALL_DIRS):
 	mkdir -p $@
 
+manta-scripts: ./deps/manta-scripts/.git
+
 $(PROTO)$(BOOT_DIR)/setup.sh: | $(INSTALL_DIRS)
 	rm -f $@ && ln -s ../$(NAME)/scripts/firstboot.sh $@
 
@@ -146,11 +160,11 @@ $(PROTO)$(PREFIX)/scripts/%.sh: boot/%.sh | $(INSTALL_DIRS)
 $(PROTO)$(PREFIX)/templates/%: templates/% | $(INSTALL_DIRS)
 	$(INSTALL_FILE)
 
-$(PROTO)$(PREFIX)/node/bin/%: $(STAMP_NODE_PREBUILT) | $(INSTALL_DIRS)
-	rm -f $@ && cp $(NODE_INSTALL)/node/bin/$(@F) $@ && chmod 755 $@
+$(PROTO)$(PREFIX)/node/bin/%: $(INSTALL_DIRS)
+	rm -f $@ && cp $(NODE_INSTALL)/bin/$(@F) $@ && chmod 755 $@
 
-$(PROTO)$(PREFIX)/node/lib/%: $(STAMP_NODE_PREBUILT) | $(INSTALL_DIRS)
-	rm -f $@ && cp $(NODE_INSTALL)/node/lib/$(@F) $@ && chmod 755 $@
+$(PROTO)$(PREFIX)/node/lib/%: $(INSTALL_DIRS)
+	rm -f $@ && cp $(NODE_INSTALL)/lib/$(@F) $@ && chmod 755 $@
 
 $(PROTO)$(PREFIX)/cmd/%.js: cmd/%.js | $(INSTALL_DIRS)
 	$(INSTALL_FILE)
@@ -178,29 +192,25 @@ $(PROTO)$(PREFIX)/sapi_manifests/%: sapi_manifests/% | $(INSTALL_DIRS)
 $(PROTO)$(PREFIX)/smf/manifests/%.xml: smf/manifests/%.xml | $(INSTALL_DIRS)
 	$(INSTALL_FILE)
 
-#
-# Mountain Gorilla targets:
-#
 
 .PHONY: release
 release: install
 	@echo "==> Building $(RELEASE_TARBALL)"
-	cd $(PROTO) && gtar -jcf $(TOP)/$(RELEASE_TARBALL) \
+	cd $(PROTO) && gtar -I pigz -cf $(TOP)/$(RELEASE_TARBALL) \
 	    --transform='s,^[^.],root/&,' \
 	    --owner=0 --group=0 \
 	    opt
 
 .PHONY: publish
 publish: release
-	@if [[ -z "$(BITS_DIR)" ]]; then \
-		echo "error: 'BITS_DIR' must be set for 'publish' target"; \
-		exit 1; \
-	fi
-	mkdir -p $(BITS_DIR)/$(NAME)
-	cp $(RELEASE_TARBALL) $(BITS_DIR)/$(NAME)/$(RELEASE_TARBALL)
-
-
-include ./tools/mk/Makefile.deps
-include ./tools/mk/Makefile.targ
-include ./tools/mk/Makefile.node_prebuilt.targ
-include ./tools/mk/Makefile.node_modules.targ
+	mkdir -p $(ENGBLD_BITS_DIR)/$(NAME)
+	cp $(RELEASE_TARBALL) $(ENGBLD_BITS_DIR)/$(NAME)/$(RELEASE_TARBALL)
+
+check:: $(NODE_EXEC)
+
+
+include ./deps/eng/tools/mk/Makefile.deps
+include ./deps/eng/tools/mk/Makefile.targ
+include ./deps/eng/tools/mk/Makefile.node_prebuilt.targ
+include ./deps/eng/tools/mk/Makefile.agent_prebuilt.targ
+include ./deps/eng/tools/mk/Makefile.node_modules.targ
diff --git a/deps/eng b/deps/eng
new file mode 160000
index 0000000..3fb93c5
--- /dev/null
+++ b/deps/eng
@@ -0,0 +1 @@
+Subproject commit 3fb93c5f46cc857ddeec97b1e178e4d9b83f39ab
diff --git a/tools/bashstyle b/tools/bashstyle
deleted file mode 100755
index 3ea6622..0000000
--- a/tools/bashstyle
+++ /dev/null
@@ -1,175 +0,0 @@
-#!/usr/bin/env node
-/*
- * This Source Code Form is subject to the terms of the Mozilla Public
- * License, v. 2.0. If a copy of the MPL was not distributed with this
- * file, You can obtain one at http://mozilla.org/MPL/2.0/.
- */
-
-/*
- * Copyright (c) 2014, Joyent, Inc.
- */
-
-/*
- * bashstyle: check bash scripts for adherence to style guidelines, including:
- *
- *    o no lines longer than 80 characters
- *    o file does not end with a blank line
- *    o Do not use 'local' and var initialization *using a subshell* in the
- *      same statement. See
- *      <http://www.tldp.org/LDP/abs/html/localvar.html#EXITVALANOMALY01>
- *      for why not. Arguably this belongs in a separate 'bashlint'.
- *
- * Future enhancements could include:
- *    o indents consistent with respect to tabs, spaces
- *    o indents consistently sized (all are some multiple of the smallest
- *      indent, which must be a tab or 4 or 8 spaces)
- */
-
-var VERSION = '2.0.0';
-
-var mod_assert = require('assert');
-var mod_fs = require('fs');
-
-var nerrors = 0;
-
-main();
-process.exit(0);
-
-function main()
-{
-	var files = process.argv.slice(2);
-
-	if (files.length === 0) {
-		console.error('usage: %s file1 [...]',
-		    process.argv.slice(0, 2).join(' '));
-		process.exit(2);
-	}
-
-	files.forEach(checkFile);
-
-	if (nerrors != 0)
-		process.exit(1);
-}
-
-function checkFile(filename)
-{
-	var text = mod_fs.readFileSync(filename, 'utf-8');
-	var lines = text.split('\n');
-	var i;
-	var styled = false;
-	var styleStart;
-
-	mod_assert.ok(lines.length > 0);
-
-	/*
-	 * Expand tabs in each line and check for long lines.
-	 */
-	for (i = 1; i <= lines.length; i++) {
-		var line = expandTabs(lines[i - 1]);
-
-		if (i > 1 && lines[i-2].match(/# BASHSTYLED/)) {
-			continue;
-		}
-
-		if (line.match(/# BEGIN BASHSTYLED/)) {
-			styleStart = i;
-			styled = true;
-		}
-
-		if (line.match(/# END BASHSTYLED/)) {
-			if (styled != true) {
-				nerrors++;
-				console.log('%s: %d: END BASHSTYLED ' +
-				    'w/o corresponding BEGIN', filename, i);
-			}
-			styled = false;
-		}
-
-		/*JSSTYLED*/
-		if (!styled && line.match(/^\s*local\s+(\w+)\s*=.*\$\(/)) {
-			nerrors++;
-			/*JSSTYLED*/
-			var m = line.match(/^\s*local\s+(\w+)\s*=/);
-			console.log('%s: %d: declaring and setting a "local" ' +
-				'var in the same statement ' +
-				'ignores a subshell return code ' +
-				'<http://www.tldp.org/LDP/abs/html/' +
-				'localvar.html#EXITVALANOMALY01>: ' +
-				'local %s=...',
-				filename, i, m[1]);
-		}
-
-		// Regexplanation: non-[, [, space (contents) space, ], non-]
-		// groups before and after brackets to ease search/replace.
-		if (!styled && line.match(/(^|[^\[])\[(\s.+\s)\]([^\]])/)) {
-			nerrors++;
-			console.log('%s: %d: prefer [[ to [ for tests.',
-			    filename, i);
-		}
-
-		if (!styled && line.length > 80) {
-			nerrors++;
-			console.log('%s: %d: line exceeds 80 columns',
-			    filename, i);
-		}
-
-		if (!styled && line.match(/\s+$/)) {
-			nerrors++;
-			console.log('%s: %d: line ends in whitespace',
-			    filename, i);
-		}
-	}
-
-	if (styled) {
-		nerrors++;
-		console.log('%s: %d: BEGIN BASHSTYLED that does not END',
-		            filename, styleStart);
-	}
-
-
-	/*
-	 * No sane editor lets you save a file without a newline at the
-	 * very end.
-	 */
-	if (lines[lines.length - 1].length !== 0) {
-		nerrors++;
-		console.log('%s: %d: file does not end with newline',
-			filename, lines.length);
-	}
-
-	/*
-	 * Since the file will always end with a newline, the last entry of
-	 * "lines" will actually be blank.
-	 */
-	if (lines.length > 1 && lines[lines.length - 2].length === 0) {
-		nerrors++;
-		console.log('%s: %d: file ends with a blank line',
-		    filename, lines.length - 1);
-	}
-}
-
-function expandTabs(text)
-{
-	var out = '';
-	var col = 0;
-	var j, k;
-
-	for (j = 0; j < text.length; j++) {
-		if (text[j] != '\t') {
-			out += text[j];
-			col++;
-			continue;
-		}
-
-		k = 8 - (col % 8);
-		col += k;
-
-		do {
-			out += ' ';
-		} while (--k > 0);
-
-		col += k;
-	}
-
-	return (out);
-}
diff --git a/tools/download_go b/tools/download_go
deleted file mode 100644
index 69e9a3e..0000000
--- a/tools/download_go
+++ /dev/null
@@ -1,126 +0,0 @@
-#!/bin/bash
-#
-# This Source Code Form is subject to the terms of the Mozilla Public
-# License, v. 2.0. If a copy of the MPL was not distributed with this
-# file, You can obtain one at http://mozilla.org/MPL/2.0/.
-#
-
-#
-# Copyright (c) 2018, Joyent, Inc.
-#
-
-#
-# This program will download a Go toolchain for a particular system.  It
-# currently expects to find illumos builds of the toolchain on the Joyent
-# download server, where we will be placing them alongside sdcnode, etc.
-# The program is designed specifically to be used as part of a make target.
-#
-# We may revisit this once the Go project makes official builds for illumos
-# systems available in the future.
-#
-# NOTE: This program comes from the "eng" repo. It's designed to be dropped
-# into other repos as-is without requiring any modifications. If you find
-# yourself changing this file, you should instead update the original copy in
-# eng.git and then update your repo to use the new version.
-#
-
-#
-# This program accepts four arguments, in the following order:
-#
-#	GOVERSION	The version of the Go toolchain to use; e.g., "1.9.2"
-#	GOOS		The build machine operating system; e.g., "solaris"
-#	GOARCH		The build machine CPU architecture; e.g., "amd64"
-#	OUTDIR		The local directory into which the downloaded tar
-#			file will be placed.
-#
-# The program will use the provided arguments to find and download an archive
-# of the Go toolchain for use on the build machine.  The archive will be named
-# for a combination of the provided arguments; e.g.,
-# "go1.9.2.solaris-amd64.tar.bz2".  A target symbolic link will also be
-# created, with just the version number in the name; e.g., "go-1.9.2.tar.bz2".
-# If the archive could not be downloaded, an error message will be printed and
-# the output file and target link will be unaffected.
-#
-
-BASEURL='https://download.joyent.com/pub/build/go/adhoc/'
-GOVERSION=$1
-GOOS=$2
-GOARCH=$3
-OUTDIR=$4
-
-if [[ -z $GOVERSION || -z $GOOS || -z $GOARCH || -z $OUTDIR ]]; then
-	printf 'ERROR: usage: download_go GOVERSION GOOS GOARCH OUTDIR\n' 2>&1
-	exit 1
-fi
-
-if [[ ! -d $OUTDIR ]]; then
-	printf 'ERROR: output directory "%s" does not exist\n' "$OUTDIR" 2>&1
-	exit 1
-fi
-
-TARGET="go-$GOVERSION.tar.bz2"
-
-#
-# Download the index page which lists the current set of available go
-# builds:
-#
-if ! list=$(curl -sSfL "$BASEURL") || [[ -z "$list" ]]; then
-	printf 'ERROR: could not download index page\n' >&2
-	exit 1
-fi
-
-#
-# Using only commonly found household items, extract the full name of the
-# go tar archive we need.  This program needs to be able to operate in a
-# minimally populated build zone, so we avoid using anything beyond basic
-# UNIX tools like "awk".
-#
-# One word to describe this process might be "brittle".
-#
-if ! name=$(/usr/bin/awk -v "v=$GOVERSION" -v "o=$GOOS" -v "a=$GOARCH" -F\" '
-    BEGIN { pattern = "^go"v"."o"-"a".tar.bz2$"; }
-    $1 == "<a href=" && $2 ~ pattern { print $2 }' <<< "$list") ||
-    [[ -z "$name" ]]; then
-	printf 'ERROR: could not locate file name in index page\n' >&2
-	printf '\t(Does Go version %s (%s-%s) exist?)\n' \
-	    "$GOVERSION" "$GOOS" "$GOARCH" >&2
-	exit 1
-fi
-
-
-#
-# If the full file name of the latest go build does not exist, download it now
-# to a temporary file.  If it succeeds, move it into place.
-#
-output_file="$OUTDIR/$name"
-if [[ ! -f $output_file ]]; then
-	printf 'Downloading Go: %s\n' "$BASEURL$name"
-
-	temp_file="$OUTDIR/.tmp.$name.$$"
-	rm -f "$temp_file"
-
-	if ! curl -sSf -o "$temp_file" "$BASEURL$name"; then
-		printf 'ERROR: could not download go\n' >&2
-		rm -f "$temp_file"
-		exit 1
-	fi
-
-	if ! mv "$temp_file" "$output_file"; then
-		printf 'ERROR: could not move tar file into place\n' >&2
-		rm -f "$temp_file"
-		exit 1
-	fi
-fi
-
-#
-# Make sure the target link points at the correct file:
-#
-rm -f "$OUTDIR/$TARGET"
-if ! ln -s "$name" "$OUTDIR/$TARGET"; then
-	printf 'ERROR: could not create target link\n' >&2
-	exit 1
-fi
-
-exit 0
-
-# vim: set ts=8 sts=8 sw=8 noet:
diff --git a/tools/download_sdcnode b/tools/download_sdcnode
deleted file mode 100755
index 214c2e1..0000000
--- a/tools/download_sdcnode
+++ /dev/null
@@ -1,77 +0,0 @@
-#!/bin/bash
-#
-# This Source Code Form is subject to the terms of the Mozilla Public
-# License, v. 2.0. If a copy of the MPL was not distributed with this
-# file, You can obtain one at http://mozilla.org/MPL/2.0/.
-#
-
-#
-# Copyright (c) 2017, Joyent, Inc.
-#
-
-BASEURL='https://download.joyent.com/pub/build/sdcnode'
-BRANCH='master'
-IMAGE=$1
-VARIANT=$2
-OUTDIR=$3
-
-if [[ -z $IMAGE || -z $VARIANT || -z $OUTDIR ]]; then
-	printf 'ERROR: usage: download_sdcnode IMAGE VARIANT OUTDIR\n' 2>&1
-	exit 1
-fi
-
-FULLURL="$BASEURL/$IMAGE/$BRANCH-latest/sdcnode/"
-TARGET="sdcnode-$VARIANT-$IMAGE.tar.gz"
-
-#
-# Download the index page which lists the current set of available sdcnode
-# builds:
-#
-if ! list=$(curl -sSfL "$FULLURL") || [[ -z "$list" ]]; then
-	printf 'ERROR: could not download index page\n' >&2
-	exit 1
-fi
-
-#
-# Using only commonly found household items, extract the full name of the
-# sdcnode tar archive we need.
-#
-# One word to describe this process might be "brittle".
-#
-if ! name=$(awk -v "v=$VARIANT" -v "b=$BRANCH" -v "i=$IMAGE" -F\" '
-    BEGIN { pattern = "^sdcnode-"v"-"i"-"b"-.*.tgz$"; }
-    $1 == "<a href=" && $2 ~ pattern { print $2 }' <<< "$list") ||
-    [[ -z "$name" ]]; then
-	printf 'ERROR: could not locate file name in index page\n' >&2
-	exit 1
-fi
-
-if [[ ! -f $OUTDIR/$name ]]; then
-	#
-	# If the full file name of the latest sdcnode build does not exist,
-	# download it now to a temporary file.  If it succeeds, move it into
-	# place.
-	#
-	if ! curl -sSf -o "$OUTDIR/.tmp.$name" "$FULLURL$name"; then
-		printf 'ERROR: could not download sdcnode' >&2
-		rm -f "$OUTPUT.tmp"
-		exit 1
-	fi
-
-	if ! mv "$OUTDIR/.tmp.$name" "$OUTDIR/$name"; then
-		exit 1
-	fi
-fi
-
-#
-# Make sure the target link points at the correct file:
-#
-rm -f "$OUTDIR/$TARGET"
-if ! ln -s "$name" "$OUTDIR/$TARGET"; then
-	printf 'ERROR: could not create target link\n' >&2
-	exit 1
-fi
-
-exit 0
-
-# vim: set ts=8 sts=8 sw=8 noet:
diff --git a/tools/jsl.node.conf b/tools/jsl.node.conf
deleted file mode 100644
index c44e8ee..0000000
--- a/tools/jsl.node.conf
+++ /dev/null
@@ -1,139 +0,0 @@
-#
-# Configuration File for JavaScript Lint 
-#
-# This configuration file can be used to lint a collection of scripts, or to enable
-# or disable warnings for scripts that are linted via the command line.
-#
-
-### Warnings
-# Enable or disable warnings based on requirements.
-# Use "+WarningName" to display or "-WarningName" to suppress.
-#
-+ambiguous_else_stmt          # the else statement could be matched with one of multiple if statements (use curly braces to indicate intent
-+ambiguous_nested_stmt        # block statements containing block statements should use curly braces to resolve ambiguity
-+ambiguous_newline            # unexpected end of line; it is ambiguous whether these lines are part of the same statement
-+anon_no_return_value         # anonymous function does not always return value
-+assign_to_function_call      # assignment to a function call
--block_without_braces         # block statement without curly braces
-+comma_separated_stmts        # multiple statements separated by commas (use semicolons?)
-+comparison_type_conv         # comparisons against null, 0, true, false, or an empty string allowing implicit type conversion (use === or !==)
-+default_not_at_end           # the default case is not at the end of the switch statement
-+dup_option_explicit          # duplicate "option explicit" control comment
-+duplicate_case_in_switch     # duplicate case in switch statement
-+duplicate_formal             # duplicate formal argument {name}
-+empty_statement              # empty statement or extra semicolon
-+identifier_hides_another     # identifer {name} hides an identifier in a parent scope
--inc_dec_within_stmt          # increment (++) and decrement (--) operators used as part of greater statement
-+incorrect_version            # Expected /*jsl:content-type*/ control comment. The script was parsed with the wrong version.
-+invalid_fallthru             # unexpected "fallthru" control comment
-+invalid_pass                 # unexpected "pass" control comment
-+jsl_cc_not_understood        # couldn't understand control comment using /*jsl:keyword*/ syntax
-+leading_decimal_point        # leading decimal point may indicate a number or an object member
-+legacy_cc_not_understood     # couldn't understand control comment using /*@keyword@*/ syntax
-+meaningless_block            # meaningless block; curly braces have no impact
-+mismatch_ctrl_comments       # mismatched control comment; "ignore" and "end" control comments must have a one-to-one correspondence
-+misplaced_regex              # regular expressions should be preceded by a left parenthesis, assignment, colon, or comma
-+missing_break                # missing break statement
-+missing_break_for_last_case  # missing break statement for last case in switch
-+missing_default_case         # missing default case in switch statement
-+missing_option_explicit      # the "option explicit" control comment is missing
-+missing_semicolon            # missing semicolon
-+missing_semicolon_for_lambda # missing semicolon for lambda assignment
-+multiple_plus_minus          # unknown order of operations for successive plus (e.g. x+++y) or minus (e.g. x---y) signs
-+nested_comment               # nested comment
-+no_return_value              # function {name} does not always return a value
-+octal_number                 # leading zeros make an octal number
-+parseint_missing_radix       # parseInt missing radix parameter
-+partial_option_explicit      # the "option explicit" control comment, if used, must be in the first script tag
-+redeclared_var               # redeclaration of {name}
-+trailing_comma_in_array      # extra comma is not recommended in array initializers
-+trailing_decimal_point       # trailing decimal point may indicate a number or an object member
-+undeclared_identifier        # undeclared identifier: {name}
-+unreachable_code             # unreachable code
--unreferenced_argument        # argument declared but never referenced: {name}
--unreferenced_function        # function is declared but never referenced: {name}
-+unreferenced_variable        # variable is declared but never referenced: {name}
-+unsupported_version          # JavaScript {version} is not supported
-+use_of_label                 # use of label
-+useless_assign               # useless assignment
-+useless_comparison           # useless comparison; comparing identical expressions
--useless_quotes               # the quotation marks are unnecessary
-+useless_void                 # use of the void type may be unnecessary (void is always undefined)
-+var_hides_arg                # variable {name} hides argument
-+want_assign_or_call          # expected an assignment or function call
-+with_statement               # with statement hides undeclared variables; use temporary variable instead
-
-
-### Output format
-# Customize the format of the error message.
-#    __FILE__ indicates current file path
-#    __FILENAME__ indicates current file name
-#    __LINE__ indicates current line
-#    __COL__ indicates current column
-#    __ERROR__ indicates error message (__ERROR_PREFIX__: __ERROR_MSG__)
-#    __ERROR_NAME__ indicates error name (used in configuration file)
-#    __ERROR_PREFIX__ indicates error prefix
-#    __ERROR_MSG__ indicates error message
-#
-# For machine-friendly output, the output format can be prefixed with
-# "encode:". If specified, all items will be encoded with C-slashes.
-#
-# Visual Studio syntax (default):
-+output-format __FILE__(__LINE__): __ERROR__
-# Alternative syntax:
-#+output-format __FILE__:__LINE__: __ERROR__
-
-
-### Context
-# Show the in-line position of the error.
-# Use "+context" to display or "-context" to suppress.
-#
-+context
-
-
-### Control Comments
-# Both JavaScript Lint and the JScript interpreter confuse each other with the syntax for
-# the /*@keyword@*/ control comments and JScript conditional comments. (The latter is
-# enabled in JScript with @cc_on@). The /*jsl:keyword*/ syntax is preferred for this reason,
-# although legacy control comments are enabled by default for backward compatibility.
-#
--legacy_control_comments
-
-
-### Defining identifiers
-# By default, "option explicit" is enabled on a per-file basis.
-# To enable this for all files, use "+always_use_option_explicit"
--always_use_option_explicit
-
-# Define certain identifiers of which the lint is not aware.
-# (Use this in conjunction with the "undeclared identifier" warning.)
-#
-# Common uses for webpages might be:
-+define __dirname
-+define clearInterval
-+define clearTimeout
-+define console
-+define exports
-+define global
-+define module
-+define process
-+define require
-+define setInterval
-+define setImmediate
-+define setTimeout
-+define Buffer
-+define JSON
-+define Math
-
-### JavaScript Version
-# To change the default JavaScript version:
-#+default-type text/javascript;version=1.5
-#+default-type text/javascript;e4x=1
-
-### Files
-# Specify which files to lint
-# Use "+recurse" to enable recursion (disabled by default).
-# To add a set of files, use "+process FileName", "+process Folder\Path\*.js",
-# or "+process Folder\Path\*.htm".
-#
-
diff --git a/tools/jsl.web.conf b/tools/jsl.web.conf
deleted file mode 100644
index b135000..0000000
--- a/tools/jsl.web.conf
+++ /dev/null
@@ -1,137 +0,0 @@
-#
-# Configuration File for JavaScript Lint 
-# Developed by Matthias Miller (http://www.JavaScriptLint.com)
-#
-# This configuration file can be used to lint a collection of scripts, or to enable
-# or disable warnings for scripts that are linted via the command line.
-#
-
-### Warnings
-# Enable or disable warnings based on requirements.
-# Use "+WarningName" to display or "-WarningName" to suppress.
-#
-+ambiguous_else_stmt          # the else statement could be matched with one of multiple if statements (use curly braces to indicate intent
-+ambiguous_nested_stmt        # block statements containing block statements should use curly braces to resolve ambiguity
-+ambiguous_newline            # unexpected end of line; it is ambiguous whether these lines are part of the same statement
-+anon_no_return_value         # anonymous function does not always return value
-+assign_to_function_call      # assignment to a function call
--block_without_braces         # block statement without curly braces
-+comma_separated_stmts        # multiple statements separated by commas (use semicolons?)
-+comparison_type_conv         # comparisons against null, 0, true, false, or an empty string allowing implicit type conversion (use === or !==)
-+default_not_at_end           # the default case is not at the end of the switch statement
-+dup_option_explicit          # duplicate "option explicit" control comment
-+duplicate_case_in_switch     # duplicate case in switch statement
-+duplicate_formal             # duplicate formal argument {name}
-+empty_statement              # empty statement or extra semicolon
-+identifier_hides_another     # identifer {name} hides an identifier in a parent scope
-+inc_dec_within_stmt          # increment (++) and decrement (--) operators used as part of greater statement
-+incorrect_version            # Expected /*jsl:content-type*/ control comment. The script was parsed with the wrong version.
-+invalid_fallthru             # unexpected "fallthru" control comment
-+invalid_pass                 # unexpected "pass" control comment
-+jsl_cc_not_understood        # couldn't understand control comment using /*jsl:keyword*/ syntax
-+leading_decimal_point        # leading decimal point may indicate a number or an object member
-+legacy_cc_not_understood     # couldn't understand control comment using /*@keyword@*/ syntax
-+meaningless_block            # meaningless block; curly braces have no impact
-+mismatch_ctrl_comments       # mismatched control comment; "ignore" and "end" control comments must have a one-to-one correspondence
-+misplaced_regex              # regular expressions should be preceded by a left parenthesis, assignment, colon, or comma
-+missing_break                # missing break statement
-+missing_break_for_last_case  # missing break statement for last case in switch
-+missing_default_case         # missing default case in switch statement
-+missing_option_explicit      # the "option explicit" control comment is missing
-+missing_semicolon            # missing semicolon
-+missing_semicolon_for_lambda # missing semicolon for lambda assignment
-+multiple_plus_minus          # unknown order of operations for successive plus (e.g. x+++y) or minus (e.g. x---y) signs
-+nested_comment               # nested comment
-+no_return_value              # function {name} does not always return a value
-+octal_number                 # leading zeros make an octal number
-+parseint_missing_radix       # parseInt missing radix parameter
-+partial_option_explicit      # the "option explicit" control comment, if used, must be in the first script tag
-+redeclared_var               # redeclaration of {name}
-+trailing_comma_in_array      # extra comma is not recommended in array initializers
-+trailing_decimal_point       # trailing decimal point may indicate a number or an object member
-+undeclared_identifier        # undeclared identifier: {name}
-+unreachable_code             # unreachable code
-+unreferenced_argument        # argument declared but never referenced: {name}
-+unreferenced_function        # function is declared but never referenced: {name}
-+unreferenced_variable        # variable is declared but never referenced: {name}
-+unsupported_version          # JavaScript {version} is not supported
-+use_of_label                 # use of label
-+useless_assign               # useless assignment
-+useless_comparison           # useless comparison; comparing identical expressions
-+useless_quotes               # the quotation marks are unnecessary
-+useless_void                 # use of the void type may be unnecessary (void is always undefined)
-+var_hides_arg                # variable {name} hides argument
-+want_assign_or_call          # expected an assignment or function call
-+with_statement               # with statement hides undeclared variables; use temporary variable instead
-
-
-### Output format
-# Customize the format of the error message.
-#    __FILE__ indicates current file path
-#    __FILENAME__ indicates current file name
-#    __LINE__ indicates current line
-#    __COL__ indicates current column
-#    __ERROR__ indicates error message (__ERROR_PREFIX__: __ERROR_MSG__)
-#    __ERROR_NAME__ indicates error name (used in configuration file)
-#    __ERROR_PREFIX__ indicates error prefix
-#    __ERROR_MSG__ indicates error message
-#
-# For machine-friendly output, the output format can be prefixed with
-# "encode:". If specified, all items will be encoded with C-slashes.
-#
-# Visual Studio syntax (default):
-+output-format __FILE__(__LINE__): __ERROR__
-# Alternative syntax:
-#+output-format __FILE__:__LINE__: __ERROR__
-
-
-### Context
-# Show the in-line position of the error.
-# Use "+context" to display or "-context" to suppress.
-#
-+context
-
-
-### Control Comments
-# Both JavaScript Lint and the JScript interpreter confuse each other with the syntax for
-# the /*@keyword@*/ control comments and JScript conditional comments. (The latter is
-# enabled in JScript with @cc_on@). The /*jsl:keyword*/ syntax is preferred for this reason,
-# although legacy control comments are enabled by default for backward compatibility.
-#
--legacy_control_comments
-
-
-### Defining identifiers
-# By default, "option explicit" is enabled on a per-file basis.
-# To enable this for all files, use "+always_use_option_explicit"
-+always_use_option_explicit
-
-# Define certain identifiers of which the lint is not aware.
-# (Use this in conjunction with the "undeclared identifier" warning.)
-#
-# Common uses for webpages might be:
-+define	JSON
-+define Math
-+define $
-+define XMLHttpRequest
-+define alert
-+define clearInterval
-+define clearTimeout
-+define confirm
-+define document
-+define setInterval
-+define setTimeout
-+define window
-
-### JavaScript Version
-# To change the default JavaScript version:
-#+default-type text/javascript;version=1.5
-#+default-type text/javascript;e4x=1
-
-### Files
-# Specify which files to lint
-# Use "+recurse" to enable recursion (disabled by default).
-# To add a set of files, use "+process FileName", "+process Folder\Path\*.js",
-# or "+process Folder\Path\*.htm".
-#
-
diff --git a/tools/jsstyle.conf b/tools/jsstyle.conf
deleted file mode 100644
index 2d13d5d..0000000
--- a/tools/jsstyle.conf
+++ /dev/null
@@ -1,14 +0,0 @@
-#
-# This Source Code Form is subject to the terms of the Mozilla Public
-# License, v. 2.0. If a copy of the MPL was not distributed with this
-# file, You can obtain one at http://mozilla.org/MPL/2.0/.
-#
-
-#
-# Copyright (c) 2017, Joyent, Inc.
-#
-
-unparenthesized-return=1
-indent=tab
-line-length=80
-literal-string-quote=single
diff --git a/tools/mk/Makefile.defs b/tools/mk/Makefile.defs
deleted file mode 100644
index f6e9042..0000000
--- a/tools/mk/Makefile.defs
+++ /dev/null
@@ -1,105 +0,0 @@
-#
-# This Source Code Form is subject to the terms of the Mozilla Public
-# License, v. 2.0. If a copy of the MPL was not distributed with this
-# file, You can obtain one at http://mozilla.org/MPL/2.0/.
-#
-
-#
-# Copyright (c) 2017, Joyent, Inc.
-#
-
-#
-# Makefile.defs: common defines.
-#
-# NOTE: This makefile comes from the "eng" repo. It's designed to be dropped
-# into other repos as-is without requiring any modifications. If you find
-# yourself changing this file, you should instead update the original copy in
-# eng.git and then update your repo to use the new version.
-#
-# This makefile defines some useful defines. Include it at the top of
-# your Makefile.
-#
-# Definitions in this Makefile:
-#
-#	TOP 		The absolute path to the project directory. The top dir.
-#	BRANCH 		The current git branch.
-#	TIMESTAMP	The timestamp for the build. This can be set via
-#			the TIMESTAMP envvar (used by MG-based builds).
-#	STAMP		A build stamp to use in built package names.
-#
-#	MAKE_STAMPS_DIR	The directory in which make stamp files are to be
-#			created.  See comments below on expensive targets.
-#
-#	CACHE_DIR	A directory tree in which temporary files may be
-#			collected by download, tarball extraction, etc.  This
-#			directory is completely removed by "make distclean".
-#			Files in this directory are not intended to be shipped.
-#
-
-TOP := $(shell pwd)
-
-#
-# Mountain Gorilla-spec'd versioning.
-# See "Package Versioning" in MG's README.md:
-# <https://mo.joyent.com/mountain-gorilla/blob/master/README.md#L139-200>
-#
-# Need GNU awk for multi-char arg to "-F".
-_AWK := $(shell (which gawk >/dev/null && echo gawk) \
-	|| (which nawk >/dev/null && echo nawk) \
-	|| echo awk)
-BRANCH := $(shell git symbolic-ref HEAD | $(_AWK) -F/ '{print $$3}')
-ifeq ($(TIMESTAMP),)
-	TIMESTAMP := $(shell date -u "+%Y%m%dT%H%M%SZ")
-endif
-_GITDESCRIBE := g$(shell git describe --all --long --dirty | $(_AWK) -F'-g' '{print $$NF}')
-STAMP := $(BRANCH)-$(TIMESTAMP)-$(_GITDESCRIBE)
-
-# node-gyp will print build info useful for debugging with V=1
-export V=1
-
-CACHE_DIR ?=		cache
-DISTCLEAN_FILES +=	$(CACHE_DIR)
-
-#
-# EXPENSIVE TARGETS AND MAKE STAMP FILES
-#
-# Targets which are expensive to run and lack a single file that marks
-# completion are difficult to track with make; e.g., "npm install".  One
-# solution to this problem is to create "stamp" files with symbolic names which
-# are created as the final step in a complex make rule in order to mark
-# completion.
-#
-# In order to make these more uniform, and easier to target with "make clean",
-# we will attempt to store them under a single directory.  Note that these
-# files are never targets for shipping in build artefacts.
-#
-# Stamp-backed targets come in several parts.  First, a macro should be defined
-# which names a file in the MAKE_STAMPS_DIR directory.  Then, a target which
-# creates this stamp file must be provided.  The recipe for this target should
-# use MAKE_STAMP_REMOVE and MAKE_STAMP_CREATE to perform the appropriate stamp
-# management.
-#
-# For example:
-#
-# --- Makefile.*.defs:
-#
-#	$(STAMP_EXPENSIVE_RESULT) := $(MAKE_STAMPS_DIR)/expensive-result
-#
-# --- Makefile.*.targ:
-#
-#	$(STAMP_EXPENSIVE_RESULT): input/file another/input/file
-#		$(MAKE_STAMP_REMOVE)
-#		rm -rf output_tree/  # <--- ensure a clean slate
-#		expensive_program -o output_tree/ $^
-#		$(MAKE_STAMP_CREATE)
-#
-# NOTE: Every stamp file is exposed as an implicit "stamp-$STAMP_NAME" target.
-# The example above could be built manually by invoking:
-#
-#	make stamp-expensive-result
-#
-MAKE_STAMPS_DIR ?=	make_stamps
-CLEAN_FILES +=		$(MAKE_STAMPS_DIR)
-
-MAKE_STAMP_REMOVE =	mkdir -p $(@D); rm -f $(@)
-MAKE_STAMP_CREATE =	mkdir -p $(@D); touch $(@)
diff --git a/tools/mk/Makefile.deps b/tools/mk/Makefile.deps
deleted file mode 100644
index 1cffbe7..0000000
--- a/tools/mk/Makefile.deps
+++ /dev/null
@@ -1,54 +0,0 @@
-# -*- mode: makefile -*-
-#
-# This Source Code Form is subject to the terms of the Mozilla Public
-# License, v. 2.0. If a copy of the MPL was not distributed with this
-# file, You can obtain one at http://mozilla.org/MPL/2.0/.
-#
-
-#
-# Copyright (c) 2014, Joyent, Inc.
-#
-
-#
-# Makefile.deps: Makefile for including common tools as dependencies
-#
-# NOTE: This makefile comes from the "eng" repo. It's designed to be dropped
-# into other repos as-is without requiring any modifications. If you find
-# yourself changing this file, you should instead update the original copy in
-# eng.git and then update your repo to use the new version.
-#
-# This file is separate from Makefile.targ so that teams can choose
-# independently whether to use the common targets in Makefile.targ and the
-# common tools here.
-#
-
-#
-# javascriptlint
-#
-JSL_EXEC	?= deps/javascriptlint/build/install/jsl
-JSL		?= $(JSL_EXEC)
-
-$(JSL_EXEC): | deps/javascriptlint/.git
-	cd deps/javascriptlint && make install
-
-distclean::
-	if [[ -f deps/javascriptlint/Makefile ]]; then \
-		cd deps/javascriptlint && make clean; \
-	fi
-
-#
-# jsstyle
-#
-JSSTYLE_EXEC	?= deps/jsstyle/jsstyle
-JSSTYLE		?= $(JSSTYLE_EXEC)
-
-$(JSSTYLE_EXEC): | deps/jsstyle/.git
-
-#
-# restdown
-#
-RESTDOWN_EXEC	?= deps/restdown/bin/restdown
-RESTDOWN	?= python $(RESTDOWN_EXEC)
-$(RESTDOWN_EXEC): | deps/restdown/.git
-
-EXTRA_DOC_DEPS	?=
diff --git a/tools/mk/Makefile.node_modules.defs b/tools/mk/Makefile.node_modules.defs
deleted file mode 100644
index ec8cc8e..0000000
--- a/tools/mk/Makefile.node_modules.defs
+++ /dev/null
@@ -1,68 +0,0 @@
-#
-# This Source Code Form is subject to the terms of the Mozilla Public
-# License, v. 2.0. If a copy of the MPL was not distributed with this
-# file, You can obtain one at http://mozilla.org/MPL/2.0/.
-#
-
-#
-# Copyright (c) 2017, Joyent, Inc.
-#
-
-#
-# Makefile.node_modules.defs: Makefile for using NPM modules.
-#
-# NOTE: This makefile comes from the "eng" repo. It's designed to be dropped
-# into other repos as-is without requiring any modifications. If you find
-# yourself changing this file, you should instead update the original copy in
-# eng.git and then update your repo to use the new version.
-#
-
-#
-# This Makefile provides a target for building NPM modules from the dependency
-# information in the "package.json" file.  The "npm install" operation is
-# expensive and produces a complex (multi-file) result which is difficult for
-# make to use in dependency analysis.  As such, we use a "stamp" file to track
-# successful completion of module installation.
-#
-# This variable allows the consumer to influence the environment used to run
-# NPM commands.
-#
-#	NPM_ENV			This string should be set to a list of
-#				environment variables in the syntax used
-#				by bash; e.g.,
-#
-#					NPM_ENV =	TESTING=yes V=1
-#
-# Consumers should, for targets which depend on the installation of NPM
-# modules, depend on the stamp file using the $(STAMP_NODE_MODULES) variable,
-# e.g.:
-#
-#	.PHONY: all
-#	all: $(STAMP_NODE_MODULES)
-#
-# A phony target, "make stamp-node-modules", is also provided to allow the
-# engineer to manually perform NPM module installation without invoking other
-# targets.  Note that this target should _not_ be used as a dependency for
-# other targets in consuming Makefiles; using phony targets to represent
-# intermediate build stages can inhibit the ability of make to determine
-# when no additional actions are required.
-#
-
-TOP ?= $(error You must include Makefile.defs before this makefile)
-NPM ?= $(error You must include either Makefile.node.defs or \
-    Makefile.node_prebuilt.defs before this makefile)
-
-BUILD ?=		build
-
-#
-# Invoking "npm install" at the top-level will create a "node_modules"
-# directory into which NPM modules will be installed.
-#
-CLEAN_FILES +=		node_modules
-
-#
-# To avoid repeatedly reinstalling from NPM, we create a "stamp" file to track
-# successful runs of "npm install".  Note that MAKE_STAMPS_DIR is included
-# in CLEAN_FILES already.
-#
-STAMP_NODE_MODULES ?=	$(MAKE_STAMPS_DIR)/node-modules
diff --git a/tools/mk/Makefile.node_modules.targ b/tools/mk/Makefile.node_modules.targ
deleted file mode 100644
index 8091a9b..0000000
--- a/tools/mk/Makefile.node_modules.targ
+++ /dev/null
@@ -1,31 +0,0 @@
-#
-# This Source Code Form is subject to the terms of the Mozilla Public
-# License, v. 2.0. If a copy of the MPL was not distributed with this
-# file, You can obtain one at http://mozilla.org/MPL/2.0/.
-#
-
-#
-# Copyright (c) 2017, Joyent, Inc.
-#
-
-#
-# Makefile.node_modules.targ: See comments in Makefile.node_modules.defs.
-#
-# NOTE: This makefile comes from the "eng" repo. It's designed to be dropped
-# into other repos as-is without requiring any modifications. If you find
-# yourself changing this file, you should instead update the original copy in
-# eng.git and then update your repo to use the new version.
-#
-
-STAMP_NODE_MODULES ?= $(error You must include Makefile.node_modules.defs \
-    before this file)
-
-#
-# If the "package.json" file changes, we need to rebuild the contents of
-# the "node_modules" directory.
-#
-$(STAMP_NODE_MODULES): package.json $(STAMP_NODE_PREBUILT) | $(NPM_EXEC)
-	$(MAKE_STAMP_REMOVE)
-	rm -rf node_modules
-	$(NPM_ENV) $(NPM) install
-	$(MAKE_STAMP_CREATE)
diff --git a/tools/mk/Makefile.node_prebuilt.defs b/tools/mk/Makefile.node_prebuilt.defs
deleted file mode 100644
index e1306ca..0000000
--- a/tools/mk/Makefile.node_prebuilt.defs
+++ /dev/null
@@ -1,96 +0,0 @@
-#
-# This Source Code Form is subject to the terms of the Mozilla Public
-# License, v. 2.0. If a copy of the MPL was not distributed with this
-# file, You can obtain one at http://mozilla.org/MPL/2.0/.
-#
-
-#
-# Copyright (c) 2017, Joyent, Inc.
-#
-
-#
-# Makefile.node_prebuilt.defs: Makefile for obtaining a prebuilt Node toolchain.
-#
-# NOTE: This makefile comes from the "eng" repo. It's designed to be dropped
-# into other repos as-is without requiring any modifications. If you find
-# yourself changing this file, you should instead update the original copy in
-# eng.git and then update your repo to use the new version.
-#
-
-#
-# This Makefile aids in the download and operation of a Node toolchain when
-# building software written in the Node language.  It provides as much isolation
-# from the broader build host as possible.
-#
-# This Makefile is intended to be used along with "Makefile.node_prebuilt.targ".
-#
-# When using this Makefile, you MUST define these variables before the include
-# directive:
-#
-#	NODE_PREBUILT_VERSION	The version of the Node toolchain to download
-#				and use.  For example, "v4.8.5".
-#
-#	NODE_PREBUILT_TAG	The type of Node toolchain build to download.
-#				For example, "gz", "zone", or "zone64".
-#
-#	NODE_PREBUILT_IMAGE	The UUID of the image in which the Node
-#				toolchain is to have been built.
-#
-# This Makefile provides definitions for the following variables:
-#
-#	NODE_INSTALL		The location of the Node toolchain, relative
-#				to $(TOP).
-#
-
-ifndef TOP
-$(error You must include Makefile.defs before this makefile.)
-endif
-
-ifndef CACHE_DIR
-$(error You must include Makefile.defs before this makefile.)
-endif
-
-ifndef NODE_PREBUILT_VERSION
-$(error NODE_PREBUILT_VERSION must be set before including this makefile.)
-endif
-
-#
-# Use a trick to create a variable that contains a single space:
-#
-EMPTY :=
-SPACE :=			$(EMPTY) $(EMPTY)
-
-NODE_VERSION =			$(NODE_PREBUILT_VERSION)
-
-NODE_ID =			$(subst $(SPACE),-,$(NODE_PREBUILT_VERSION) \
-				$(NODE_PREBUILT_TAG) $(NODE_PREBUILT_IMAGE))
-
-#
-# The "tools/download_go" script will obtain a Node toolchain tar archive, which
-# we will store in the $(CACHE_DIR).  This directory is be removed entirely by
-# "make distclean".
-#
-NODE_TARBALL =			$(CACHE_DIR)/sdcnode-$(NODE_ID).tar.gz
-
-#
-# The downloaded Go toolchain will be extracted into a directory under
-# $(CACHE_DIR) by the $(STAMP_GO_TOOLCHAIN) target.  This directory becomes
-# the value of $GOROOT for toolchain invocations.
-#
-NODE_INSTALL =			$(CACHE_DIR)/sdcnode-$(NODE_ID)
-NODE_BIN_DIR =			$(NODE_INSTALL)/node/bin
-
-#
-# If the version of Go is changed in the Makefile, or interactively, we need
-# to make sure the new version is downloaded and installed.  As such, the
-# stamp name needs to include the version.
-#
-STAMP_NODE_PREBUILT =		$(MAKE_STAMPS_DIR)/sdcnode-$(NODE_ID)
-
-CLEAN_FILES +=			$(NODE_INSTALL)
-CLEAN_FILES +=			$(NODE_DEV_SYMLINK)
-
-NODE =				$(TOP)/$(NODE_BIN_DIR)/node
-NODE_WAF =			$(TOP)/$(NODE_BIN_DIR)/node-waf
-NPM =				PATH=$(TOP)/$(NODE_BIN_DIR):$(PATH) \
-				    $(NODE) $(TOP)/$(NODE_BIN_DIR)/npm
diff --git a/tools/mk/Makefile.node_prebuilt.targ b/tools/mk/Makefile.node_prebuilt.targ
deleted file mode 100644
index 72caf68..0000000
--- a/tools/mk/Makefile.node_prebuilt.targ
+++ /dev/null
@@ -1,48 +0,0 @@
-#
-# This Source Code Form is subject to the terms of the Mozilla Public
-# License, v. 2.0. If a copy of the MPL was not distributed with this
-# file, You can obtain one at http://mozilla.org/MPL/2.0/.
-#
-
-#
-# Copyright (c) 2017, Joyent, Inc.
-#
-
-#
-# Makefile.node_prebuilt.targ: Makefile for obtaining a prebuilt Node toolchain.
-#
-# NOTE: This makefile comes from the "eng" repo. It's designed to be dropped
-# into other repos as-is without requiring any modifications. If you find
-# yourself changing this file, you should instead update the original copy in
-# eng.git and then update your repo to use the new version.
-#
-
-#
-# For usage documentation, see the comments in "Makefile.node_prebuilt.defs".
-#
-
-ifndef NODE_TARBALL
-$(error You must include Makefile.node_prebuilt.defs first.)
-endif
-
-#
-# Download a prebuilt copy of the Node toolchain.
-#
-$(NODE_TARBALL):
-	rm -f $@
-	mkdir -p $(@D)
-	tools/download_sdcnode $(NODE_PREBUILT_IMAGE) \
-	    $(NODE_PREBUILT_VERSION)-$(NODE_PREBUILT_TAG) $(@D)
-
-#
-# Extract the Node toolchain.  This stamp includes the version number of the
-# Node toolchain, ensuring a new download and extraction if the version changes.
-#
-$(STAMP_NODE_PREBUILT): $(NODE_TARBALL)
-	$(MAKE_STAMP_REMOVE)
-	rm -rf $(NODE_INSTALL)
-	mkdir $(NODE_INSTALL)
-	cd $(NODE_INSTALL) && tar xfz $(TOP)/$(NODE_TARBALL)
-	if [[ -n "$(NODE_DEV_SYMLINK)" ]]; then rm -f $(NODE_DEV_SYMLINK); \
-	    ln -s $(NODE_INSTALL)/node $(NODE_DEV_SYMLINK); fi
-	$(MAKE_STAMP_CREATE)
diff --git a/tools/mk/Makefile.targ b/tools/mk/Makefile.targ
deleted file mode 100644
index 54edf0e..0000000
--- a/tools/mk/Makefile.targ
+++ /dev/null
@@ -1,334 +0,0 @@
-#
-# This Source Code Form is subject to the terms of the Mozilla Public
-# License, v. 2.0. If a copy of the MPL was not distributed with this
-# file, You can obtain one at http://mozilla.org/MPL/2.0/.
-#
-
-#
-# Copyright (c) 2017, Joyent, Inc.
-#
-
-#
-# Makefile.targ: common targets.
-#
-# NOTE: This makefile comes from the "eng" repo. It's designed to be dropped
-# into other repos as-is without requiring any modifications. If you find
-# yourself changing this file, you should instead update the original copy in
-# eng.git and then update your repo to use the new version.
-#
-# This Makefile defines several useful targets and rules. You can use it by
-# including it from a Makefile that specifies some of the variables below.
-#
-# Targets defined in this Makefile:
-#
-#	check	Checks JavaScript files for lint and style
-#		Checks bash scripts for syntax
-#		Checks SMF manifests for validity against the SMF DTD
-#
-#	clean	Removes built files
-#
-#	docs	Builds restdown documentation in docs/
-#
-#	prepush	Depends on "check" and "test"
-#
-#	test	Does nothing (you should override this)
-#
-#	xref	Generates cscope (source cross-reference index)
-#
-# For details on what these targets are supposed to do, see the Joyent
-# Engineering Guide.
-#
-# To make use of these targets, you'll need to set some of these variables. Any
-# variables left unset will simply not be used.
-#
-#	BASH_FILES	Bash scripts to check for syntax
-#			(paths relative to top-level Makefile)
-#
-#	CLEAN_FILES	Files to remove as part of the "clean" target.  Note
-#			that files generated by targets in this Makefile are
-#			automatically included in CLEAN_FILES.  These include
-#			restdown-generated HTML and JSON files.
-#
-#	DOC_FILES	Restdown (documentation source) files. These are
-#			assumed to be contained in "docs/", and must NOT
-#			contain the "docs/" prefix.
-#
-#	JSL_CONF_NODE	Specify JavaScriptLint configuration files
-#	JSL_CONF_WEB	(paths relative to top-level Makefile)
-#
-#			Node.js and Web configuration files are separate
-#			because you'll usually want different global variable
-#			configurations.  If no file is specified, none is given
-#			to jsl, which causes it to use a default configuration,
-#			which probably isn't what you want.
-#
-#	JSL_FILES_NODE	JavaScript files to check with Node config file.
-#	JSL_FILES_WEB	JavaScript files to check with Web config file.
-#
-#	JSON_FILES	JSON files to be validated
-#
-#	JSSTYLE_FILES	JavaScript files to be style-checked
-#
-# You can also override these variables:
-#
-#	BASH		Path to bash (default: "bash")
-#
-#	CSCOPE_DIRS	Directories to search for source files for the cscope
-#			index. (default: ".")
-#
-#	JSL		Path to JavaScriptLint (default: "jsl")
-#
-#	JSL_FLAGS_NODE	Additional flags to pass through to JSL
-#	JSL_FLAGS_WEB
-#	JSL_FLAGS
-#
-#	JSON		Path to json tool (default: "json")
-#
-#	JSSTYLE		Path to jsstyle (default: "jsstyle")
-#
-#	JSSTYLE_FLAGS	Additional flags to pass through to jsstyle
-#
-#	RESTDOWN_EXT	By default '.md' is required for DOC_FILES (see above).
-#			If you want to use, say, '.restdown' instead, then set
-#			'RESTDOWN_EXT=.restdown' in your Makefile.
-#
-
-#
-# Defaults for the various tools we use.
-#
-BASH		?= bash
-BASHSTYLE	?= tools/bashstyle
-CP		?= cp
-CSCOPE		?= cscope
-CSCOPE_DIRS	?= .
-JSL		?= jsl
-JSON		?= json
-JSSTYLE		?= jsstyle
-MKDIR		?= mkdir -p
-MV		?= mv
-RESTDOWN_FLAGS	?=
-RESTDOWN_EXT	?= .md
-RMTREE		?= rm -rf
-JSL_FLAGS  	?= --nologo --nosummary
-
-ifeq ($(shell uname -s),SunOS)
-	TAR	?= gtar
-else
-	TAR	?= tar
-endif
-
-
-#
-# Defaults for other fixed values.
-#
-BUILD		= build
-DISTCLEAN_FILES += $(BUILD)
-DOC_BUILD	= $(BUILD)/docs/public
-
-#
-# Configure JSL_FLAGS_{NODE,WEB} based on JSL_CONF_{NODE,WEB}.
-#
-ifneq ($(origin JSL_CONF_NODE), undefined)
-	JSL_FLAGS_NODE += --conf=$(JSL_CONF_NODE)
-endif
-
-ifneq ($(origin JSL_CONF_WEB), undefined)
-	JSL_FLAGS_WEB += --conf=$(JSL_CONF_WEB)
-endif
-
-#
-# Targets. For descriptions on what these are supposed to do, see the
-# Joyent Engineering Guide.
-#
-
-#
-# Instruct make to keep around temporary files. We have rules below that
-# automatically update git submodules as needed, but they employ a deps/*/.git
-# temporary file. Without this directive, make tries to remove these .git
-# directories after the build has completed.
-#
-.SECONDARY: $($(wildcard deps/*):%=%/.git)
-
-#
-# This rule enables other rules that use files from a git submodule to have
-# those files depend on deps/module/.git and have "make" automatically check
-# out the submodule as needed.
-#
-deps/%/.git:
-	git submodule update --init deps/$*
-
-#
-# These recipes make heavy use of dynamically-created phony targets. The parent
-# Makefile defines a list of input files like BASH_FILES. We then say that each
-# of these files depends on a fake target called filename.bashchk, and then we
-# define a pattern rule for those targets that runs bash in check-syntax-only
-# mode. This mechanism has the nice properties that if you specify zero files,
-# the rule becomes a noop (unlike a single rule to check all bash files, which
-# would invoke bash with zero files), and you can check individual files from
-# the command line with "make filename.bashchk".
-#
-.PHONY: check-bash
-check-bash: $(BASH_FILES:%=%.bashchk) $(BASH_FILES:%=%.bashstyle)
-
-%.bashchk: %
-	$(BASH) -n $^
-
-%.bashstyle: %
-	$(BASHSTYLE) $^
-
-.PHONY: check-json
-check-json: $(JSON_FILES:%=%.jsonchk)
-
-%.jsonchk: %
-	$(JSON) --validate -f $^
-
-#
-# The above approach can be slow when there are many files to check because it
-# requires that "make" invoke the check tool once for each file, rather than
-# passing in several files at once.  For the JavaScript check targets, we define
-# a variable for the target itself *only if* the list of input files is
-# non-empty.  This avoids invoking the tool if there are no files to check.
-#
-JSL_NODE_TARGET = $(if $(JSL_FILES_NODE), check-jsl-node)
-.PHONY: check-jsl-node
-check-jsl-node: $(JSL_EXEC)
-	$(JSL) $(JSL_FLAGS) $(JSL_FLAGS_NODE) $(JSL_FILES_NODE)
-
-JSL_WEB_TARGET = $(if $(JSL_FILES_WEB), check-jsl-web)
-.PHONY: check-jsl-web
-check-jsl-web: $(JSL_EXEC)
-	$(JSL) $(JSL_FLAGS) $(JSL_FLAGS_WEB) $(JSL_FILES_WEB)
-
-.PHONY: check-jsl
-check-jsl: $(JSL_NODE_TARGET) $(JSL_WEB_TARGET)
-
-JSSTYLE_TARGET = $(if $(JSSTYLE_FILES), check-jsstyle)
-.PHONY: check-jsstyle
-check-jsstyle:  $(JSSTYLE_EXEC)
-	$(JSSTYLE) $(JSSTYLE_FLAGS) $(JSSTYLE_FILES)
-
-.PHONY: check
-check:: check-jsl check-json $(JSSTYLE_TARGET) check-bash
-	@echo check ok
-
-.PHONY: clean
-clean::
-	-$(RMTREE) $(CLEAN_FILES)
-
-.PHONY: distclean
-distclean:: clean
-	-$(RMTREE) $(DISTCLEAN_FILES)
-
-CSCOPE_FILES = cscope.in.out cscope.out cscope.po.out
-CLEAN_FILES += $(CSCOPE_FILES)
-
-.PHONY: xref
-xref: cscope.files
-	$(CSCOPE) -bqR
-
-.PHONY: cscope.files
-cscope.files:
-	find $(CSCOPE_DIRS) -name '*.c' -o -name '*.h' -o -name '*.cc' \
-	    -o -name '*.js' -o -name '*.s' -o -name '*.cpp' > $@
-
-#
-# The "docs" target is complicated because we do several things here:
-#
-#    (1) Use restdown to build HTML and JSON files from each of DOC_FILES.
-#
-#    (2) Copy these files into $(DOC_BUILD) (build/docs/public), which
-#        functions as a complete copy of the documentation that could be
-#        mirrored or served over HTTP.
-#
-#    (3) Then copy any directories and media from docs/media into
-#        $(DOC_BUILD)/media. This allows projects to include their own media,
-#        including files that will override same-named files provided by
-#        restdown.
-#
-# Step (3) is the surprisingly complex part: in order to do this, we need to
-# identify the subdirectories in docs/media, recreate them in
-# $(DOC_BUILD)/media, then do the same with the files.
-#
-DOC_MEDIA_DIRS := $(shell find docs/media -type d 2>/dev/null | grep -v "^docs/media$$")
-DOC_MEDIA_DIRS := $(DOC_MEDIA_DIRS:docs/media/%=%)
-DOC_MEDIA_DIRS_BUILD := $(DOC_MEDIA_DIRS:%=$(DOC_BUILD)/media/%)
-
-DOC_MEDIA_FILES := $(shell find docs/media -type f 2>/dev/null)
-DOC_MEDIA_FILES := $(DOC_MEDIA_FILES:docs/media/%=%)
-DOC_MEDIA_FILES_BUILD := $(DOC_MEDIA_FILES:%=$(DOC_BUILD)/media/%)
-
-#
-# Like the other targets, "docs" just depends on the final files we want to
-# create in $(DOC_BUILD), leveraging other targets and recipes to define how
-# to get there.
-#
-.PHONY: docs
-docs::							\
-	$(DOC_FILES:%$(RESTDOWN_EXT)=$(DOC_BUILD)/%.html)		\
-	$(DOC_FILES:%$(RESTDOWN_EXT)=$(DOC_BUILD)/%.json)		\
-	$(DOC_MEDIA_FILES_BUILD)
-
-#
-# We keep the intermediate files so that the next build can see whether the
-# files in DOC_BUILD are up to date.
-#
-.PRECIOUS:					\
-	$(DOC_FILES:%$(RESTDOWN_EXT)=docs/%.html)		\
-	$(DOC_FILES:%$(RESTDOWN_EXT)=docs/%json)
-
-#
-# We do clean those intermediate files, as well as all of DOC_BUILD.
-#
-CLEAN_FILES +=					\
-	$(DOC_BUILD)				\
-	$(DOC_FILES:%$(RESTDOWN_EXT)=docs/%.html)		\
-	$(DOC_FILES:%$(RESTDOWN_EXT)=docs/%.json)
-
-#
-# Before installing the files, we must make sure the directories exist. The |
-# syntax tells make that the dependency need only exist, not be up to date.
-# Otherwise, it might try to rebuild spuriously because the directory itself
-# appears out of date.
-#
-$(DOC_MEDIA_FILES_BUILD): | $(DOC_MEDIA_DIRS_BUILD)
-
-$(DOC_BUILD)/%: docs/% | $(DOC_BUILD)
-	$(MKDIR) $(shell dirname $@)
-	$(CP) $< $@
-
-docs/%.json docs/%.html: docs/%$(RESTDOWN_EXT) | $(DOC_BUILD) $(RESTDOWN_EXEC) \
-    $(EXTRA_DOC_DEPS)
-	$(RESTDOWN) $(RESTDOWN_FLAGS) -m $(DOC_BUILD) $<
-
-$(DOC_BUILD):
-	$(MKDIR) $@
-
-$(DOC_MEDIA_DIRS_BUILD):
-	$(MKDIR) $@
-
-#
-# The default "test" target does nothing. This should usually be overridden by
-# the parent Makefile. It's included here so we can define "prepush" without
-# requiring the repo to define "test".
-#
-.PHONY: test
-test:
-
-.PHONY: prepush
-prepush: check test
-
-#
-# This rule automatically exposes every "stamp" file as a target that can be
-# invoked manually as "stamp-$STAMP_NAME".  For example, if a stamp has been
-# defined thus:
-#
-#	STAMP_EXPENSIVE_RESULT := $(MAKE_STAMPS_DIR)/expensive-result
-#
-# ... this can be invoked manually as "make stamp-expensive-result".  Note that
-# these phony targets are essentially just for interactive usage.  Targets
-# should be specified to depend on the macro containing the stamp file name.
-#
-# See also the comments in "Makefile.defs".
-#
-stamp-%: $(MAKE_STAMPS_DIR)/%
-	@:
diff --git a/tools/mkrepo b/tools/mkrepo
deleted file mode 100644
index 3b0d624..0000000
--- a/tools/mkrepo
+++ /dev/null
@@ -1,121 +0,0 @@
-#!/usr/bin/env node
-// -*- mode: js -*-
-/*
- * This Source Code Form is subject to the terms of the Mozilla Public
- * License, v. 2.0. If a copy of the MPL was not distributed with this
- * file, You can obtain one at http://mozilla.org/MPL/2.0/.
- */
-
-/*
- * Copyright (c) 2014, Joyent, Inc.
- */
-
-var child_process = require('child_process');
-var fs = require('fs');
-var path = require('path');
-
-
-
-///--- Globals
-
-var DIRS = [
-    'deps',
-    'docs',
-    'docs/media',
-    'lib',
-    'smf',
-    'smf/manifests',
-    'test',
-    'tools',
-    'tools/mk'
-];
-
-var SUBMODULES = {
-    'javascriptlint': 'https://github.com/davepacheco/javascriptlint.git',
-    'jsstyle': 'https://github.com/davepacheco/jsstyle.git',
-    'restdown': 'https://github.com/trentm/restdown.git'
-};
-
-
-
-///--- Internal Functions
-
-function usage(code, message) {
-    if (message)
-        console.error(message);
-
-    console.error('usage: %s [repo ...]', path.basename(process.argv[1]));
-    process.exit(code);
-}
-
-
-function ensureDirectoryNotExists(dir) {
-    try {
-        var stats = fs.statSync(dir);
-        usage(1, dir + ' already exists');
-    } catch (e) {
-        return false;
-    }
-}
-
-
-function cp(src, dest) {
-    fs.createReadStream(src).pipe(fs.createWriteStream(dest));
-}
-
-
-function exec(cmd, dir, cb) {
-    child_process.exec(cmd, {cwd: dir}, function (err, stdout, stderr) {
-        if (err)
-            process.exit(err.code || 1);
-
-        if (typeof (cb) === 'function')
-            return cb(null);
-    });
-}
-
-
-function mkdir(d) {
-    fs.mkdirSync(d, '0750');
-}
-
-function gitify(dir, repo) {
-    exec('git init', dir, function () {
-        exec('git remote add origin git@github.com:joyent/' + repo + '.git',
-	    dir);
-
-        Object.keys(SUBMODULES).forEach(function (k) {
-            // stub out the git submodule call
-            console.error('Cloning into deps/' + k + '...');
-            exec('git submodule add ' + SUBMODULES[k] + ' ./deps/' + k, dir);
-        });
-    });
-}
-
-
-
-///--- Mainline
-
-if (process.argv.length < 3)
-    usage(1, 'repo required');
-
-process.argv.slice(2).forEach(function (arg) {
-    var repo = path.resolve(arg);
-    ensureDirectoryNotExists(repo);
-    mkdir(repo);
-    DIRS.concat('.').forEach(function (d) {
-        var dir = repo + '/' + d;
-        if (d != '.')
-            mkdir(dir);
-
-        fs.readdirSync('./' + d).forEach(function (f) {
-            var src = './' + d + '/' + f;
-            var dest = dir + '/' + f;
-            if (fs.statSync(src).isFile() && !/^\..*/.test(f))
-                cp(src, dest);
-        });
-    });
-
-    cp('./.gitignore', repo + '/.gitignore');
-    gitify(repo, arg);
-});
diff --git a/tools/runtests.in b/tools/runtests.in
deleted file mode 100644
index 2d72b6f..0000000
--- a/tools/runtests.in
+++ /dev/null
@@ -1,136 +0,0 @@
-#!/bin/bash
-#
-# This Source Code Form is subject to the terms of the Mozilla Public
-# License, v. 2.0. If a copy of the MPL was not distributed with this
-# file, You can obtain one at http://mozilla.org/MPL/2.0/.
-#
-
-#
-# Copyright (c) 2014, Joyent, Inc.
-#
-
-#
-# Run the TODONAME tests.
-# Run `./runtests -h` for usage info.
-#
-
-if [ "$TRACE" != "" ]; then
-    export PS4='${BASH_SOURCE}:${LINENO}: ${FUNCNAME[0]:+${FUNCNAME[0]}(): }'
-    set -o xtrace
-fi
-set -o errexit
-set -o pipefail
-
-
-
-#---- guard
-
-if [[ ! -f "/lib/sdc/.sdc-test-no-production-data" ]]; then
-    cat <<EOF
-To run this test you must create the file:
-
-    /lib/sdc/.sdc-test-no-production-data
-
-after ensuring you have no production data on this SDC.
-EOF
-    exit 2
-fi
-
-
-
-#---- config
-
-NAME=TODONAME
-TOP=$(cd $(dirname $0)/../; pwd)
-
-
-
-#---- support functions
-
-function fatal
-{
-    echo "$(basename $0): fatal error: $*"
-    exit 1
-}
-
-function usage
-{
-    echo "Usage:"
-    echo "  runtests [OPTIONS...]"
-    echo ""
-    echo "Options:"
-    echo "  -f FILTER   Filter pattern (substring match) for test files to run."
-}
-
-
-
-#---- mainline
-
-start_time=$(date +%s)
-
-# Options.
-opt_test_pattern=
-while getopts "hf:" opt
-do
-    case "$opt" in
-        h)
-            usage
-            exit 0
-            ;;
-        f)
-            opt_test_pattern=$OPTARG
-            ;;
-        *)
-            usage
-            exit 1
-            ;;
-    esac
-done
-
-OUTPUT_DIR=/var/tmp/${NAME}test
-echo "# Create output dir ($OUTPUT_DIR)."
-rm -rf $OUTPUT_DIR
-mkdir -p $OUTPUT_DIR
-
-
-
-#---- start tests
-
-# TODO
-# Project specific setup and test running goes here.
-# - TAP output should be redirected or tee'd to $OUTPUT_DIR/*.tap
-#   This is used (a) to summarize below and (b) for the Jenkins Job TAP
-#   parsing/reporting.
-# - If reasonable, use $opt_test_pattern to filter the set of test files run.
-#
-# Some examples:
-# - https://mo.joyent.com/amon/blob/master/test/runtests
-# - https://mo.joyent.com/smartos-live/blob/master/src/vm/runtests
-# TODO
-
-
-
-#---- summary
-
-echo ""
-echo "# test results:"
-
-end_time=$(date +%s)
-elapsed=$((${end_time} - ${start_time}))
-
-tests=$(grep "^# tests [0-9]" $OUTPUT_DIR/*.tap | cut -d ' ' -f3 | xargs | tr ' ' '+' | bc)
-passed=$(grep "^# pass  [0-9]" $OUTPUT_DIR/*.tap | tr -s ' ' | cut -d ' ' -f3 | xargs | tr ' ' '+' | bc)
-[[ -z ${tests} ]] && tests=0
-[[ -z ${passed} ]] && passed=0
-fail=$((${tests} - ${passed}))
-
-echo "# Completed in ${elapsed} seconds."
-echo -e "# \033[32mPASS: ${passed} / ${tests}\033[39m"
-if [[ ${fail} -gt 0 ]]; then
-    echo -e "# \033[31mFAIL: ${fail} / ${tests}\033[39m"
-fi
-echo ""
-
-if [[ ${tests} != ${passed} ]]; then
-    exit 1
-fi
diff --git a/tools/service_bundle.dtd.1 b/tools/service_bundle.dtd.1
deleted file mode 100644
index e5c2380..0000000
--- a/tools/service_bundle.dtd.1
+++ /dev/null
@@ -1,1091 +0,0 @@
-<?xml version="1.0" encoding="UTF-8"?>
-<!--
- Copyright (c) 2004, 2010, Oracle and/or its affiliates. All rights reserved.
-
- CDDL HEADER START
-
- The contents of this file are subject to the terms of the
- Common Development and Distribution License (the "License").
- You may not use this file except in compliance with the License.
-
- You can obtain a copy of the license at usr/src/OPENSOLARIS.LICENSE
- or http://www.opensolaris.org/os/licensing.
- See the License for the specific language governing permissions
- and limitations under the License.
-
- When distributing Covered Code, include this CDDL HEADER in each
- file and include the License file at usr/src/OPENSOLARIS.LICENSE.
- If applicable, add the following below this CDDL HEADER, with the
- fields enclosed by brackets "[]" replaced with your own identifying
- information: Portions Copyright [yyyy] [name of copyright owner]
-
- CDDL HEADER END
--->
-
-<!--
-  Service description DTD
-
-    Most attributes are string values (or an individual string from a
-    restricted set), but attributes with a specific type requirement are
-    noted in the comment describing the element.
--->
-
-<!--
-  XInclude support
-
-    A series of service bundles may be composed via the xi:include tag.
-    smf(5) tools enforce that all bundles be of the same type.
--->
-
-<!--
-     These entities are used for the property, propval and property_group
-     elements, that require type attributes for manifest, while for profiles
-     the type attributes are only implied.
--->
-
-<!ENTITY % profile "IGNORE">
-<!ENTITY % manifest "INCLUDE">
-
-<!ELEMENT xi:include
-  (xi:fallback)
-  >
-<!ATTLIST xi:include
-  href CDATA #REQUIRED
-  parse (xml|text) "xml"
-  encoding CDATA #IMPLIED
-  xmlns:xi CDATA #FIXED "http://www.w3.org/2001/XInclude"
-  >
-
-<!ELEMENT xi:fallback
-  ANY
-  >
-<!ATTLIST xi:fallback
-  xmlns:xi CDATA #FIXED "http://www.w3.org/2001/XInclude"
-  >
-
-<!--
-  stability
-
-    This element associates an SMI stability level with the parent
-    element.  See attributes(5) for an explanation of interface
-    stability levels.
-
-    Its attribute is
-
-	value	The stability level of the parent element.
--->
-
-<!ELEMENT stability EMPTY>
-
-<!ATTLIST stability
-	value		( Standard | Stable | Evolving | Unstable |
-			External | Obsolete ) #REQUIRED >
-
-<!-- Property value lists -->
-
-<!--
-  value_node
-
-    This element represents a single value within any of the typed
-    property value lists.
-
-    Its attribute is
-
-	value	The value for this node in the list.
--->
-
-<!ELEMENT value_node EMPTY>
-
-<!ATTLIST value_node
-	value CDATA #REQUIRED>
-
-<!--
-  count_list
-  integer_list
-  opaque_list
-  host_list
-  hostname_list
-  net_address_list
-  net_address_v4_list
-  net_address_v6_list
-  time_list
-  astring_list
-  ustring_list
-  boolean_list
-  fmri_list
-  uri_list
-
-    These elements represent the typed lists of values for a property.
-    Each contains one or more value_node elements representing each
-    value on the list.
-
-    None of these elements has attributes.
--->
-
-<!ELEMENT count_list
-	( value_node+ )>
-
-<!ATTLIST count_list>
-
-<!ELEMENT integer_list
-	( value_node+ )>
-
-<!ATTLIST integer_list>
-
-<!ELEMENT opaque_list
-	( value_node+ )>
-
-<!ATTLIST opaque_list>
-
-<!ELEMENT host_list
-	( value_node+ )>
-
-<!ATTLIST host_list>
-
-<!ELEMENT hostname_list
-	( value_node+ )>
-
-<!ATTLIST hostname_list>
-
-<!ELEMENT net_address_list
-	( value_node+ )>
-
-<!ATTLIST net_address_list>
-
-<!ELEMENT net_address_v4_list
-	( value_node+ )>
-
-<!ATTLIST net_address_v4_list>
-
-<!ELEMENT net_address_v6_list
-	( value_node+ )>
-
-<!ATTLIST net_address_v6_list>
-
-<!ELEMENT time_list
-	( value_node+ )>
-
-<!ATTLIST time_list>
-
-<!ELEMENT astring_list
-	( value_node+ )>
-
-<!ATTLIST astring_list>
-
-<!ELEMENT ustring_list
-	( value_node+ )>
-
-<!ATTLIST ustring_list>
-
-<!ELEMENT boolean_list
-	( value_node+ )>
-
-<!ATTLIST boolean_list>
-
-<!ELEMENT fmri_list
-	( value_node+ )>
-
-<!ATTLIST fmri_list>
-
-<!ELEMENT uri_list
-	( value_node+ )>
-
-<!ATTLIST uri_list>
-
-<!-- Properties and property groups -->
-
-<!--
-   property
-
-     This element is for a singly or multiply valued property within a
-     property group.  It contains an appropriate value list element,
-     which is expected to be consistent with the type attribute.
-
-     Its attributes are
-
-	name	The name of this property.
-
-	type	The data type for this property.
-
-	override These values should replace values already in the
-		repository.
--->
-
-<![%profile;[
-<!ELEMENT property
-	( count_list | integer_list | opaque_list | host_list | hostname_list |
-	net_address_list | net_address_v4_list | net_address_v6_list |
-	time_list | astring_list | ustring_list | boolean_list | fmri_list |
-	uri_list )? >
-
-<!ATTLIST property
-	name		CDATA #REQUIRED
-	type		( count | integer | opaque | host | hostname |
-			net_address | net_address_v4 | net_address_v6 | time |
-			astring | ustring | boolean | fmri | uri ) #IMPLIED
-	override	( true | false ) "false" >
-]]>
-	
-<![%manifest;[
-<!ELEMENT property
-	( count_list | integer_list | opaque_list | host_list | hostname_list |
-	net_address_list | net_address_v4_list | net_address_v6_list |
-	time_list | astring_list | ustring_list | boolean_list | fmri_list |
-	uri_list )? >
-
-<!ATTLIST property
-	name		CDATA #REQUIRED
-	type		( count | integer | opaque | host | hostname |
-			net_address | net_address_v4 | net_address_v6 | time |
-			astring | ustring | boolean | fmri | uri ) #REQUIRED
-	override	( true | false ) "false" >
-]]>
-
-<!--
-   propval
-
-     This element is for a singly valued property within a property
-     group.  List-valued properties must use the property element above.
-
-     Its attributes are
-
-	name	The name of this property.
-
-	type	The data type for this property.
-
-	value	The value for this property.  Must match type
-		restriction of type attribute.
-
-	override This value should replace any values already in the
-		repository.
--->
-
-<![%profile;[
-<!ELEMENT propval EMPTY>
-
-<!ATTLIST propval
-	name		CDATA #REQUIRED
-	type		( count | integer | opaque | host | hostname |
-			net_address | net_address_v4 | net_address_v6 | time |
-			astring | ustring | boolean | fmri | uri ) #IMPLIED
-	value		CDATA #REQUIRED
-	override	( true | false ) "false" >
-]]>
-
-<![%manifest;[
-<!ELEMENT propval EMPTY>
-
-<!ATTLIST propval
-	name		CDATA #REQUIRED
-	type		( count | integer | opaque | host | hostname |
-			net_address | net_address_v4 | net_address_v6 | time |
-			astring | ustring | boolean | fmri | uri ) #REQUIRED
-	value		CDATA #REQUIRED
-	override	( true | false ) "false" >
-]]>
-
-<!--
-  property_group
-
-    This element is for a set of related properties on a service or
-    instance.  It contains an optional stability element, as well as
-    zero or more property-containing elements.
-
-    Its attributes are
-
-	name	The name of this property group.
-
-	type	A category for this property group.  Groups of type
-		"framework", "implementation" or "template" are primarily
-		of interest to the service management facility, while
-		groups of type "application" are expected to be only of
-		interest to the service to which this group is attached.
-		Other types may be introduced using the service symbol
-		namespace conventions.
-
-	delete	If in the repository, this property group should be removed.
--->
-
-<![%profile;[
-<!ELEMENT property_group
-	( stability?, ( propval | property )* )>
-
-<!ATTLIST property_group
-	name		CDATA #REQUIRED
-	type		CDATA #IMPLIED
-	delete		( true | false ) "false" >
-]]>
-
-<![%manifest;[
-<!ELEMENT property_group
-	( stability?, ( propval | property )* )>
-
-<!ATTLIST property_group
-	name		CDATA #REQUIRED
-	type		CDATA #REQUIRED
-	delete		( true | false ) "false" >
-]]>
-
-<!--
-  service_fmri
-
-    This element defines a reference to a service FMRI (for either a
-    service or an instance).
-
-    Its attribute is
-
-	value	The FMRI.
--->
-
-<!ELEMENT service_fmri EMPTY>
-
-<!ATTLIST service_fmri
-	value		CDATA #REQUIRED>
-
-<!-- Dependencies -->
-
-<!--
-  dependency
-
-    This element identifies a group of FMRIs upon which the service is
-    in some sense dependent.  Its interpretation is left to the
-    restarter to which a particular service instance is delegated.  It
-    contains a group of service FMRIs, as well as a block of properties.
-
-    Its attributes are
-
-	name	The name of this dependency.
-
-	grouping The relationship between the various FMRIs grouped
-		here; "require_all" of the FMRIs to be online, "require_any"
-		of the FMRIs to be online, or "exclude_all" of the FMRIs
-		from being online or in maintenance for the dependency to
-		be satisfied.  "optional_all" dependencies are satisfied
-		when all of the FMRIs are either online or unable to come
-		online (because they are disabled, misconfigured, or one
-		of their dependencies is unable to come online).
-
-	restart_on The type of events from the FMRIs that the service should
-		be restarted for.  "error" restarts the service if the
-		dependency is restarted due to hardware fault.  "restart"
-		restarts the service if the dependency is restarted for
-		any reason, including hardware fault.  "refresh" restarts
-		the service if the dependency is refreshed or restarted for
-		any reason.  "none" will never restart the service due to
-		dependency state changes.
-
-	type	The type of dependency: on another service ('service'), on
-		a filesystem path ('path'), or another dependency type.
-
-	delete	This dependency should be deleted.
--->
-
-<!ELEMENT dependency
-	( service_fmri*, stability?, ( propval | property )* ) >
-
-<!ATTLIST dependency
-	name		CDATA #REQUIRED
-	grouping	( require_all | require_any | exclude_all |
-			optional_all ) #REQUIRED
-	restart_on	( error | restart | refresh | none ) #REQUIRED
-	type		CDATA #REQUIRED
-	delete		( true | false ) "false" >
-
-<!-- Dependents -->
-
-<!--
-  dependent
-
-    This element identifies a service which should depend on this service.  It
-    corresponds to a dependency in the named service.  The grouping and type
-    attributes of that dependency are implied to be "require_all" and
-    "service", respectively.
-
-    Its attributes are
-
-	name	The name of the dependency property group to create in the
-		dependent entity.
-
-	grouping The grouping relationship of the dependency property
-		group to create in the dependent entity.  See "grouping"
-		attribute on the dependency element.
-
-	restart_on The type of events from this service that the named service
-		should be restarted for.
-
-	delete	True if this dependent should be deleted.
-
-	override Whether to replace an existing dependent of the same name.
-
--->
-
-<!ELEMENT dependent
-	( service_fmri, stability?, ( propval | property )* ) >
-
-<!ATTLIST dependent
-	name		CDATA #REQUIRED
-	grouping	( require_all | require_any | exclude_all |
-			optional_all) #REQUIRED
-	restart_on	( error | restart | refresh | none) #REQUIRED
-	delete		( true | false ) "false"
-	override	( true | false ) "false" >
-
-<!-- Method execution context, security profile, and credential definitions -->
-
-<!--
-  envvar
-
-    An environment variable. It has two attributes:
-
-	name	The name of the environment variable.
-	value	The value of the environment variable.
--->
-
-<!ELEMENT envvar EMPTY>
-
-<!ATTLIST envvar
-	name		CDATA #REQUIRED
-	value		CDATA #REQUIRED >
-
-<!--
-  method_environment
-
-    This element defines the environment for a method. It has no
-    attributes, and one or more envvar child elements.
--->
-
-<!ELEMENT method_environment (envvar+) >
-
-<!ATTLIST method_environment>
-
-<!--
-  method_profile
-
-    This element indicates which exec_attr(5) profile applies to the
-    method context being defined.
-
-    Its attribute is
-
-	name	The name of the profile.
--->
-
-<!ELEMENT method_profile EMPTY>
-
-<!ATTLIST method_profile
-	name		CDATA #REQUIRED >
-
-<!--
-  method_credential
-
-    This element specifies credential attributes for the execution
-    method to use.
-
-    Its attributes are
-
-	user	The user ID, in numeric or text form.
-
-	group	The group ID, in numeric or text form.  If absent or
-		":default", the group associated with the user in the
-		passwd database.
-
-	supp_groups Supplementary group IDs to be associated with the
-		method, separated by commas or spaces.  If absent or
-		":default", initgroups(3C) will be used.
-
-	privileges An optional string specifying the privilege set.
-
-	limit_privileges An optional string specifying the limit
-		privilege set.
--->
-
-<!ELEMENT method_credential EMPTY>
-
-<!ATTLIST method_credential
-	user		CDATA #REQUIRED
-	group		CDATA #IMPLIED
-	supp_groups	CDATA #IMPLIED
-	privileges	CDATA #IMPLIED
-	limit_privileges CDATA #IMPLIED >
-
-<!--
-  method_context
-
-    This element combines credential and resource management attributes
-    for execution methods.  It may contain a method_environment, or
-    a method_profile or method_credential element.
-
-    Its attributes are
-
-	working_directory The home directory to launch the method from.
-		":default" can be used as a token to indicate use of the
-		user specified by the credential or profile specified.
-
-	project	The project ID, in numeric or text form.  ":default" can
-		be used as a token to indicate use of the project
-		identified by getdefaultproj(3PROJECT) for the non-root
-		user specified by the credential or profile specified.
-		If the user is root, ":default" designates the project
-		the restarter is running in.
-
-	resource_pool The resource pool name to launch the method on.
-		":default" can be used as a token to indicate use of the
-		pool specified in the project(4) entry given in the
-		"project" attribute above.
--->
-<!ELEMENT method_context
-	( (method_profile | method_credential)?, method_environment? ) >
-
-<!ATTLIST method_context
-	working_directory	CDATA #IMPLIED
-	project			CDATA #IMPLIED
-	resource_pool		CDATA #IMPLIED >
-
-<!-- Restarter delegation, methods, and monitors -->
-
-<!--
-  exec_method
-
-    This element describes one of the methods used by the designated
-    restarter to act on the service instance.  Its interpretation is
-    left to the restarter to which a particular service instance is
-    delegated.  It contains a set of attributes, an optional method
-    context, and an optional stability element for the optional
-    properties that can be included.
-
-    Its attributes are
-
-	type	The type of method, either "method" or "monitor".
-
-	name	Name of this execution method.  The method names are
-		usually a defined interface of the restarter to which an
-		instance of this service is delegated.
-
-	exec	The string identifying the action to take.  For
-		svc.startd(1M), this is a string suitable to pass to
-		exec(2).
-
-	timeout_seconds [integer] Duration, in seconds, to wait for this
-		method to complete.  A '0' or '-1' denotes an infinite
-		timeout.
-
-	delete	If in the repository, the property group for this method
-		should be removed.
--->
-
-<!ELEMENT exec_method
-	( method_context?, stability?, ( propval | property )* ) >
-
-<!ATTLIST exec_method
-	type		( method | monitor ) #REQUIRED
-	name		CDATA #REQUIRED
-	exec		CDATA #REQUIRED
-	timeout_seconds	CDATA #REQUIRED
-	delete		( true | false ) "false" >
-
-<!--
-  restarter
-
-    A flag element identifying the restarter to which this service or
-    service instance is delegated.  Contains the FMRI naming the
-    delegated restarter.
-
-    This element has no attributes.
--->
-
-<!ELEMENT restarter
-	( service_fmri ) >
-
-<!ATTLIST restarter>
-
-<!--
-  Templates
--->
-
-<!--
-  doc_link
-
-    The doc_link relates a resource described by the given URI to the
-    service described by the containing template.  The resource is
-    expected to be a documentation or elucidatory reference of some
-    kind.
-
-    Its attributes are
-
-      name      A label for this resource.
-
-      uri       A URI to the resource.
--->
-
-<!ELEMENT doc_link EMPTY>
-
-<!ATTLIST doc_link
-	name		CDATA #REQUIRED
-	uri		CDATA #REQUIRED >
-
-<!--
-  manpage
-
-    The manpage element connects the reference manual page to the
-    template's service.
-
-    Its attributes are
-
-      title     The manual page title.
-
-      section   The manual page's section.
-
-      manpath   The MANPATH environment variable, as described in man(1)
-                that is required to reach the named manual page
--->
-
-<!ELEMENT manpage EMPTY>
-
-<!ATTLIST manpage
-	title		CDATA #REQUIRED
-	section		CDATA #REQUIRED
-	manpath		CDATA ":default" >
-
-<!--
-  documentation
-
-    The documentation element groups an arbitrary number of doc_link
-    and manpage references.
-
-    It has no attributes.
--->
-
-<!ELEMENT documentation
-	( doc_link | manpage )* >
-
-<!ATTLIST documentation>
-
-<!--
-  loctext
-
-    The loctext element is a container for localized text.
-
-    Its sole attribute is
-
-	xml:lang The name of the locale, in the form accepted by LC_ALL,
-		etc.  See locale(5).
--->
-<!ELEMENT loctext
-        (#PCDATA) >
-
-<!ATTLIST loctext
-        xml:lang	CDATA #REQUIRED >
-
-<!--
-  description
-
-    The description holds a set of potentially longer, localized strings that
-    consist of a short description of the service.
-
-    The description has no attributes.
--->
-<!ELEMENT description
-        ( loctext+ ) >
-
-<!ATTLIST description>
-
-<!--
-  common_name
-
-    The common_name holds a set of short, localized strings that
-    represent a well-known name for the service in the given locale.
-
-    The common_name has no attributes.
--->
-<!ELEMENT common_name
-        ( loctext+ ) >
-
-<!ATTLIST common_name>
-
-<!--
-  units
-
-    The units a numerical property is expressed in.
--->
-
-<!ELEMENT units
-	( loctext+ ) >
-
-<!ATTLIST units>
-
-<!--
-  visibility
-
-    Expresses how a property is typically accessed.  This isn't
-    intended as access control, but as an indicator as to how a
-    property is used.
-
-    Its attributes are:
-
-      value     'hidden', 'readonly', or 'readwrite' indicating that
-		the property should be hidden from the user, shown but
-		read-only, or modifiable.
--->
-
-<!ELEMENT visibility EMPTY>
-
-<!ATTLIST visibility
-	value	( hidden | readonly | readwrite ) #REQUIRED >
-
-<!--
-  value
-
-    Describes a legal value for a property value, and optionally contains a
-    human-readable name and description for the specified property
-    value.
-
-    Its attributes are:
-
-      name	A string representation of the value.
--->
-
-<!ELEMENT value
-	( common_name?, description? ) >
-
-<!ATTLIST value
-	name	CDATA #REQUIRED >
-
-<!--
-  values
-
-    Human-readable names and descriptions for valid values of a property.
--->
-
-<!ELEMENT values
-	(value+) >
-
-<!ATTLIST values>
-
-<!--
-  cardinality
-
-    Places a constraint on the number of values the property can take
-    on.
-
-    Its attributes are:
-	min	minimum number of values
-	max	maximum number of values
-
-    Both attributes are optional.  If min is not specified, it defaults to
-    0.  If max is not specified it indicates an unlimited number of values.
-    If neither is specified this indicates 0 or more values.
--->
-
-<!ELEMENT cardinality EMPTY>
-
-<!ATTLIST cardinality
-	min	CDATA "0"
-	max	CDATA "18446744073709551615">
-
-<!--
-  internal_separators
-
-    Indicates the separators used within a property's value used to
-    separate the actual values.  Used in situations where multiple
-    values are packed into a single property value instead of using a
-    multi-valued property.
--->
-
-<!ELEMENT internal_separators
-	(#PCDATA) >
-
-<!ATTLIST internal_separators>
-
-<!--
-  range
-
-    Indicates a range of possible integer values.
-
-    Its attributes are:
-
-      min	The minimum value of the range (inclusive).
-      max	The maximum value of the range (inclusive).
--->
-
-<!ELEMENT range EMPTY>
-
-<!ATTLIST range
-	min	CDATA #REQUIRED
-	max	CDATA #REQUIRED >
-
-<!--
-  constraints
-
-    Provides a set of constraints on the values a property can take on.
--->
-
-<!ELEMENT constraints
-	( value*, range* ) >
-<!ATTLIST constraints>
-
-<!--
-  include_values
-
-    Includes an entire set of values in the choices block.
-
-    Its attributes are:
-
-	type    Either "constraints" or "values", indicating an
-		inclusion of all values allowed by the property's
-		constraints or all values for which there are
-		human-readable names and descriptions, respectively.
--->
-
-<!ELEMENT include_values EMPTY>
-
-<!ATTLIST include_values
-	type	( constraints | values ) #REQUIRED >
-
-<!--
-  choices
-
-    Provides a set of common choices for the values a property can take
-    on.  Useful in those cases where the possibilities are unenumerable
-    or merely inconveniently legion, and a manageable subset is desired
-    for presentation in a user interface.
--->
-
-<!ELEMENT choices
-	( value*, range*, include_values* ) >
-
-<!ATTLIST choices>
-
-<!--
-  prop_pattern
-
-
-    The prop_pattern describes one property of the enclosing property group
-    pattern.
-
-    Its attributes are:
-
-	name    The property's name.
-	type    The property's type.
-	required
-		If the property group is present, this property is required.
-
-	type can be omitted if required is false.
--->
-
-<!ELEMENT prop_pattern
-	( common_name?, description?, units?, visibility?, cardinality?,
-	  internal_separators?, values?, constraints?, choices? ) >
-
-<!ATTLIST prop_pattern
-	name		CDATA	#REQUIRED
-	type		( count | integer | opaque | host | hostname |
-			net_address | net_address_v4 | net_address_v6 | time |
-			astring | ustring | boolean | fmri | uri ) #IMPLIED
-	required	( true | false )	"false" >
-
-<!--
-  pg_pattern
-
-    The pg_pattern describes one property group.
-    Depending on the element's attributes, these descriptions may apply
-    to just the enclosing service/instance, instances of the enclosing
-    service, delegates of the service (assuming it is a restarter), or
-    all services.
-
-    Its attributes are:
-
-	name    The property group's name.  If not specified, it
-		matches all property groups with the specified type.
-	type    The property group's type.  If not specified, it
-		matches all property groups with the specified name.
-	required
-		If the property group is required.
-	target	The scope of the pattern, which may be all, delegate,
-		instance, or this.  'all' is reserved for framework use
-		and applies the template to all services on the system.
-		'delegate' is reserved for restarters, and means the
-		template applies to all services which use the restarter.
-		'this' would refer to the defining service or instance.
-		'instance' can only be used in a service's template block,
-		and means the definition applies to all instances of this
-		service.
-
--->
-
-<!ELEMENT pg_pattern
-	( common_name?, description?, prop_pattern* ) >
-
-<!ATTLIST pg_pattern
-	name		CDATA	""
-	type		CDATA	""
-	required	( true | false )	"false"
-	target		( this | instance | delegate | all )	"this" >
-
-<!--
-  template
-
-    The template contains a collection of metadata about the service.
-    It contains a localizable string that serves as a common,
-    human-readable name for the service.  (This name should be less than
-    60 characters in a single byte locale.)  The template may optionally
-    contain a longer localizable description of the service, a
-    collection of links to documentation, either in the form of manual
-    pages or in the form of URI specifications to external documentation
-    sources (such as docs.sun.com).
-
-    The template has no attributes.
--->
-<!ELEMENT template
-        ( common_name, description?, documentation?, pg_pattern* ) >
-
-<!ATTLIST template>
-
-<!-- Notification Parameters -->
-
-<!ELEMENT paramval EMPTY>
-
-<!ATTLIST paramval
-	name		CDATA #REQUIRED
-	value		CDATA #REQUIRED>
-
-<!ELEMENT parameter
-	( value_node* )>
-
-<!ATTLIST parameter
-	name		CDATA #REQUIRED>
-
-<!ELEMENT event EMPTY>
-
-<!ATTLIST event
-	value		CDATA #REQUIRED>
-
-<!ELEMENT type
-	( ( parameter | paramval )* )>
-
-<!ATTLIST type
-	name		CDATA #REQUIRED
-	active		( true | false ) "true" >
-
-<!--
-  notification parameters
-
-    This element sets the notification parameters for Software Events and
-    Fault Management problem lifecycle events.
--->
-
-<!ELEMENT notification_parameters
-	( event, type+ )>
-
-<!ATTLIST notification_parameters>
-
-<!-- Services and instances -->
-
-<!--
-  create_default_instance
-
-    A flag element indicating that an otherwise empty default instance
-    of this service (named "default") should be created at install, with
-    its enabled property set as given.
-
-    Its attribute is
-
-	enabled	[boolean] The initial value for the enabled state of
-		this instance.
--->
-
-<!ELEMENT create_default_instance EMPTY >
-
-<!ATTLIST create_default_instance
-	enabled		( true | false ) #REQUIRED >
-
-<!--
-  single_instance
-
-    A flag element stating that this service can only have a single
-    instance on a particular system.
--->
-
-<!ELEMENT single_instance EMPTY>
-
-<!ATTLIST single_instance>
-
-<!--
-  instance
-
-    The service instance is the object representing a software component
-    that will run on the system if enabled.  It contains an enabled
-    element, a set of dependencies on other services, potentially
-    customized methods or configuration data, an optional method
-    context, and a pointer to its restarter.  (If no restarter is
-    specified, the master restarter, svc.startd(1M), is assumed to be
-    responsible for the service.)
-
-    Its attributes are
-
-	name	The canonical name for this instance of the service.
-
-	enabled	[boolean] The initial value for the enabled state of
-		this instance.
--->
-
-<!ELEMENT instance
-	( restarter?, dependency*, dependent*, method_context?,
-	exec_method*, notification_parameters*, property_group*,
-	template? ) >
-
-<!ATTLIST instance
-	name		CDATA #REQUIRED
-	enabled		( true | false ) #REQUIRED >
-
-<!--
-  service
-
-    The service contains the set of instances defined by default for
-    this service, an optional method execution context, any default
-    methods, the template, and various restrictions or advice applicable
-    at installation.  The method execution context and template elements
-    are required for service_bundle documents with type "manifest", but
-    are optional for "profile" or "archive" documents.
-
-    Its attributes are
-
-	name	The canonical name for the service.
-
-	version	[integer] The integer version for this service.
-
-	type	Whether this service is a simple service, a delegated
-		restarter, or a milestone (a synthetic service that
-		collects a group of dependencies).
--->
-
-<!ELEMENT service
-	( create_default_instance?, single_instance?, restarter?,
-	dependency*, dependent*, method_context?, exec_method*,
-	notification_parameters*, property_group*, instance*,
-	stability?, template? ) >
-
-<!ATTLIST service
-	name		CDATA #REQUIRED
-	version		CDATA #REQUIRED
-	type		( service | restarter | milestone ) #REQUIRED >
-
-<!--
-  service_bundle
-
-    The bundle possesses two attributes:
-
-	type	How this file is to be understood by the framework (or
-		used in a non-framework compliant way). Standard types
-		are 'archive', 'manifest', and 'profile'.
-	
-	name	A name for the bundle.  Manifests should be named after
-		the package which delivered them; profiles should be
-		named after the "feature set nickname" they intend to
-		enable.
--->
-
-<!ELEMENT service_bundle
-	( service_bundle* | service* | xi:include* )>
-
-<!ATTLIST service_bundle
-	type		CDATA #REQUIRED
-	name		CDATA #REQUIRED>
